enterprise/frontend/src/embedding-sdk/components/private/QuestionEditor/QuestionEditor.tsx

```diff
@@ -0,0 +1,65 @@
+import { useDisclosure } from "@mantine/hooks";
+import { useState } from "react";
+
+import type { InteractiveQuestionProps } from "embedding-sdk/components/public/InteractiveQuestion";
+import { InteractiveQuestion } from "embedding-sdk/components/public/InteractiveQuestion";
+import { Box, Group, Overlay, Paper, Tabs } from "metabase/ui";
+
+export const QuestionEditor = ({
+  questionId,
+  plugins,
+}: InteractiveQuestionProps) => {
+  const [activeTab, setActiveTab] = useState<
+    "notebook" | "visualization" | (string & unknown) | null
+  >("notebook");
+  const [isSaveFormOpen, { open: openSaveForm, close: closeSaveForm }] =
+    useDisclosure(false);
+
+  return (
+    <InteractiveQuestion questionId={questionId} plugins={plugins}>
+      <Box w="100%" h="100%">
+        <Tabs
+          value={activeTab}
+          onTabChange={setActiveTab}
+          defaultValue="notebook"
+        >
+          <Group position="apart">
+            <Group>
+              <Tabs.Tab value="notebook">Notebook</Tabs.Tab>
+              <Tabs.Tab value="visualization">Visualization</Tabs.Tab>
```

Comment:
there is still a problem with `CreateQuestion` when clicking this tab without clicking the visualize button first.
![image](https://github.com/user-attachments/assets/f8a3a957-c860-4c02-b37b-b26d56ef2197)

Reply:
Added some logic to hide the visualization tab. also added some logic to ensure that we run the query when we move to the tab so the visualization is always up-to-date